algosdk/logic.py

```diff
@@ -12,8 +13,54 @@
 opcodes = None
 
 
+def _sanity_check_program(program):
+    """
+    Performs heuristic program validation:
+    check if passed in bytes are Algorand address, or they are B64 encoded, rather than Teal bytes
+
+    Args:
+        program (bytes): compiled program
+    """
+
+    def is_ascii_printable(program_bytes):
+        return all(
+            map(
+                lambda x: x == ord("\n") or (ord(" ") <= x <= ord("~")),
+                program_bytes,
+            )
+        )
+
+    if not program:
+        raise error.InvalidProgram("empty program")
+
+    if is_ascii_printable(program):
+        try:
+            encoding.decode_address(program.decode("utf-8"))
+            raise error.InvalidProgram(
+                "requesting program bytes, get Algorand address"
+            )
+        except error.WrongChecksumError:
+            pass
+        except error.WrongKeyLengthError:
+            pass
+
+        try:
+            base64.b64decode(program.decode("utf-8"))
+            raise error.InvalidProgram("program should not be b64 encoded")
+        except binascii.Error:
+            pass
+
+        raise error.InvalidProgram(
+            "program bytes are all ASCII printable characters, not looking like Teal byte code"
+        )
+
+
 def check_program(program, args=None):
     """
+    NOTE: This class is deprecated:
```

Comment:
@ahangsu Not a fully formed thought, but I've run out of time for this review cycle:  When reviewing Python-specific deprecation methods did you assess https://docs.python.org/3/library/exceptions.html#DeprecationWarning?

Looking for your assessment based on what I've found.

* From my partial read, the warning _is_ the Python-way of adding a deprecation warning.  Whether or not the docstring should _also_ be updated is unclear to me.
* I'm also noting there is a prior example in the repo using the warning: https://github.com/algorand/py-algorand-sdk/blob/9f496246124090157770da4b6dc012f5de0e1275/algosdk/mnemonic.py#L93-L97
* I began reading https://docs.python.org/3/library/warnings.html, but haven't finished.  It explains how/where the warning appears.  
  * For example, the example warning I shared above appears in the build:  https://app.circleci.com/pipelines/github/algorand/py-algorand-sdk/890/workflows/a84d0e9a-76ce-470a-8109-e32112fce5a1/jobs/4012?invite=true#step-104-22.

Reply:
makes sense, I look into the `DeprecationWarning` and will add a similar line in `mnemonic.py` in these deprecated methods.

Reply:
now the `DeprecationWarning` is reflected in CI build log: https://app.circleci.com/pipelines/github/algorand/py-algorand-sdk/894/workflows/5422bf31-4108-4c90-8260-e71b32bfd3b4/jobs/4042

Reply:
@ahangsu Thanks for the change - resolving.

For completeness - It's unclear to me if we still also need (or want) to mention deprecation in the docstring.  Since it's already been done, I'm opting to suggest no further changes.